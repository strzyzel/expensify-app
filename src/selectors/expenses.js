import moment from 'moment';
export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAt = moment(expense.createdAt);
        const startDateMatch = startDate ? createdAt.isSameOrAfter(startDate, 'day') : true;
        const endDateMatch = endDate ? createdAt.isSameOrBefore(endDate, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if(sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if(sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
};