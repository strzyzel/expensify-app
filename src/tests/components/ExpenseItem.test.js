import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseItem';
import { expenses } from '../fixtures/expenses';
test('should render Expense Item correctly', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});