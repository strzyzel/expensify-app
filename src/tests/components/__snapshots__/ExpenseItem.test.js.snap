// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense Item correctly 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <p>
      Description: 
      Gum
    </p>
  </Link>
  <p>
    Amount: 
    195
  </p>
  <p>
    Created at: 
    1970-01-01T00:00:00.000Z
  </p>
</div>
`;
